packages/markerikson-stack-client-prototype/src/app/api.ts

```diff
@@ -0,0 +1,147 @@
+import { createApi, fakeBaseQuery } from "@reduxjs/toolkit/query/react";
+import type {
+  newSource,
+  ProtocolClient,
+  Location,
+  PointDescription,
+  HitCount,
+  createPauseResult,
+} from "@replayio/protocol";
+import { Dictionary } from "lodash";
+import groupBy from "lodash/groupBy";
+
+import { client, initSocket } from "protocol/socket";
+import { replayClient } from "../client/ReplayClient";
+
+interface SourceGroups {
+  src: newSource[];
+  node_modules: newSource[];
+  other: newSource[];
+}
+
+const reIsJsSourceFile = /(js|ts)x?$/;
+
+const CACHE_DATA_PERMANENTLY = 10 * 365 * 24 * 60 * 60;
+
+export const api = createApi({
+  baseQuery: fakeBaseQuery(),
```

Comment:
I've used RTK Query for interacting with REST API's, and it is *awesome*. We definitely get *some* of the awesomeness by coercing it to work with the websocket client, but it does feel like an awkward fit to me.

Reply:
Agreed. This bit feels a little awkward to me as well. Not a _huge_ concern though.

Reply:
You could also use an identity function or just skip the `baseQuery` here (although TS would not be happy) - the benefit of the `fakeBaseQuery` is that it allows you to specify a custom error type.

Reply:
You can use `queryFn` either way, `fakeBaseQuery` just throws an error if you accidentally use `query` instead of `queryFn` and allows you to specify an error type for TS.